### PR TITLE
add build tasks for www

### PR DESCRIPTION
**Pull request by arlimus**

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @arlimus

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

File: tasks/www.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'inquirer'
+
+namespace :www do
+  desc 'Builds the site locally'
+  task :build do
+    system('cd www/ && middleman build')
```
Comment (chris-rock) [inline tasks/www.rb]: should we use bundler? or do we expect that users have middleman installed?

File: tasks/www.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    unless File.directory?(dst) && File.file?(File.join(dst, 'index.html'))
+      fail 'It looks like the siste was not build. Aborting.'
+    end
+
+    # check if git exists
+    system('command -v git >/dev/null 2>&1') ||
+      fail("It looks like `git` isn't installed. It is required to run this build task.")
+
+    unless system('git diff-index --quiet HEAD --')
```
Comment (chris-rock) [inline tasks/www.rb:45]: 👍 for all the checks

File: tasks/www.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      fail 'Please make sure you have no uncommitted changes in this repository.'
+    end
+
+    file_count = Dir['www/build/*'].length
+    file_size = `du -hs www/build`.sub(/\s+.*$/m, '')
+
+    puts '----> Remove local gh-pages branch'
+    system('git branch -D gh-pages')
```
Comment (chris-rock) [inline tasks/www.rb:54]: Shouldn't we backup at least the last commit?

File: tasks/www.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+#
+
+require 'inquirer'
+
+namespace :www do
+  desc 'Builds the site locally'
+  task :build do
+    Bundler.with_clean_env {
+      system('cd www/ && bundle install && bundle exec middleman build')
+    }
```
Comment (arlimus) [inline tasks/www.rb:25]: bundler within bundler... inception anyone?

**Review by chris-rock — APPROVED** [review]

File: tasks/www.rb (hunk 3 above)
Comment (arlimus) [reply tasks/www.rb:54]: That's a great idea! Let's add a mechanism after this MR

added to https://github.com/chef/inspec/issues/1114 for tracking

**Pull request merged**